api/src/main/java/com/epam/esm/controller/TagController.java

```diff
@@ -0,0 +1,64 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.model.ErrorResponse;
+import com.epam.esm.model.dto.TagDto;
+import com.epam.esm.service.TagService;
+import io.swagger.annotations.ApiOperation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@Controller
+@RequestMapping("/api/v1/tags")
+public class TagController {
+    private final TagService tagService;
+
+    @Autowired
+    public TagController(TagService tagService) {
+        this.tagService = tagService;
+    }
+
+    @GetMapping
+    @ApiOperation(value = "Api v1. Get all tags")
+    public ResponseEntity<List<TagDto>> allTags(@RequestParam(required = false) String name) {
+        List<TagDto> tags;
+
+        if (name != null){
+            tags = tagService.getTagByPartName(name);
+        }else {
```

Comment:
CTRL+ALT+L here

Reply:
can be refactored as 
`List<TagDto> tags = name != null
? tagService.getTagByPartName(name)
: tags = tagService.getAllTags();`